### PR TITLE
A Concourse Resource for puppet forge.

### PR DESCRIPTION
**Pull request by ndmckinley**
This is just the code for the Concourse Resource for puppet forge - it's got `in`, where you can grab a module from puppet forge, it's got `check`, which gives you the list, and it's got `out`, where you can publish to puppet forge.

`out` doesn't _by default_ commit the modification to the version number to git, but you can easily do that if appropriate for your workflow, since it _does_ modify the file in-place.

Question: what _else_ do we need to be ready to release 0.3.0?

-----------------------------------------------------------------
# [all]
CI change only.

### CODE REVIEW

File: .ci/containers/puppet-forge/out.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  patch = 0
+  minor += 1
+end
+
+metadata = JSON.parse(File.open(File.join(ARGV[0], 'metadata.json')).read)
+unless metadata['name'] == config['source']['module_name']
+  raise "Cowardly refusing to push #{metadata['name']} to #{config['source']['module_name']}"
+end
+metadata['version'] = "#{major}.#{minor}.#{patch}"
+File.write(File.join(ARGV[0], 'metadata.json'), JSON.dump(metadata))
```
Comment (rambleraptor) [inline .ci/containers/puppet-forge/out.rb]: Check for ARGV[0] and document what value it's holding.

File: .ci/containers/puppet-forge/in.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require 'puppet_forge'
+require 'json'
+
+config = JSON.parse(STDIN.read)
+raise 'You need to define `module_name`' unless config['source'].key? 'module_name'
+raise 'Surprised by lack of ARGV[0]' if ARGV[0].nil?
```
Comment (rambleraptor) [inline .ci/containers/puppet-forge/in.rb]: What is ARGV[0] in this context?

File: .ci/containers/puppet-forge/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,64 @@
+from alpine:latest as resource
```
Comment (rambleraptor) [inline .ci/containers/puppet-forge/Dockerfile]: Could we use the Ruby container we already have?

File: .ci/containers/puppet-forge/in.rb (hunk 2 above)
Comment (ndmckinley) [reply .ci/containers/puppet-forge/in.rb]: Documented!  It's the directory to put stuff in.

File: .ci/containers/puppet-forge/out.rb (hunk 1 above)
Comment (ndmckinley) [reply .ci/containers/puppet-forge/out.rb]: Done!

File: .ci/containers/puppet-forge/Dockerfile (hunk 3 above)
Comment (ndmckinley) [reply .ci/containers/puppet-forge/Dockerfile]: Eh... we *could*, but I don't really want to.  The one we already have is go + ruby.  This is a little different, it's git + ruby.  This separates concerns in a way I find aesthetically appealing - if we break the terraform generation container, we don't also have to break puppet forge pushes.

**Pull request merged**